force-app/main/default/classes/EmailUtility.cls

```diff
@@ -0,0 +1,36 @@
+public class EmailUtility {
+    public static void toSendEmailWithTemplate(String Name, String objectType){
+        List<Messaging.SingleEmailMessage> mailList = new List<Messaging.SingleEmailMessage>();
+        
+        EmailTemplate emailTemp = [SELECT Id,
+                                  Subject,
+                                  Body
+                                  FROM EmailTemplate
+                                  WHERE DeveloperName = 'Decision_Maker'];
+        
+        try{
+            sObject targetObject;
+        if(objectType == 'Contact'){
+            targetObject = [SELECT Id, Email FROM Contact WHERE LastName =: Name];
+        }else if(objectType == 'User'){
+            targetObject = [SELECT Id, Email FROM User WHERE LastName =: Name];
+        }
+        
+        Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
+        mail.setSaveAsActivity(false);
+        mail.setTemplateId(emailTemp.Id);
+        mail.setTargetObjectId(targetObject.Id);
+        mailList.add(mail);
+        
+        if(!mailList.isEmpty())
+            Messaging.sendEmail(mailList);
+        }catch(QueryException qe) {
+            System.debug('There is no contacts or opportunities with this lastname:'+qe.getMessage());
+        }catch(NullPointerException npe){
```

Comment:
better to check above if object is null instead of allowing to throw NullPointer exception. All possible nulls references should be handled to prevent NullPointerException